hermes/probe/delete/delete_file.go

```diff
@@ -0,0 +1,143 @@
+//		- FileMissing: the file to be deleted could not be found in the target bucket.
+//		- BucketMissing: the target bucket on this target system was not found.
+//		- ProbeFailed: there was an error during one of the API calls and the probe failed.
+func DeleteFile(ctx context.Context, fileID int32, target *probe.Target, client stiface.Client, logger *logger.Logger) (int32, error) {
+	bucket := target.Target.GetBucketName()
+
+	// TODO(evanSpendlove): Add custom error object to return value and modify all returns.
+	filename, ok := target.Journal.Filenames[fileID]
+	if !ok {
+		return fileID, fmt.Errorf("delete(%q, %q) failed; status StateJournalInconsistent: expected fileID %d to exist", bucket, filename, fileID)
+	}
+
+	target.Journal.Intent = &pb.Intent{
+		FileOperation: pb.Intent_CREATE,
+		Filename:      filename,
+	}
+
+	file := client.Bucket(bucket).Object(filename)
+
+	start := time.Now()
+	if err := file.Delete(ctx); err != nil {
+		var status m.ExitStatus
+		switch err {
+		case storage.ErrObjectNotExist:
+			status = m.FileMissing
+		case storage.ErrBucketNotExist:
+			status = m.BucketMissing
+		default:
+			status = m.ProbeFailed
+		}
+
+		target.LatencyMetrics.APICallLatency[m.APIDeleteFile][status].Metric("hermes_api_latency_s").AddFloat64(time.Now().Sub(start).Seconds())
+		return fileID, fmt.Errorf("delete(%q, %q) failed; status %v: %w", bucket, filename, status, err)
+	}
+	target.LatencyMetrics.APICallLatency[m.APIDeleteFile][m.Success].Metric("hermes_api_latency_s").AddFloat64(time.Now().Sub(start).Seconds())
+
+	query := &storage.Query{Prefix: filename}
+	start = time.Now()
+	objects := client.Bucket(bucket).Objects(ctx, query)
+	target.LatencyMetrics.APICallLatency[m.APIListFiles][m.Success].Metric("hermes_api_latency_s").AddFloat64(time.Now().Sub(start).Seconds())
+	for {
+		obj, err := objects.Next()
+		if err == iterator.Done {
+			break
+		}
+		if obj.Name == filename {
+			status := m.ProbeFailed
+			return fileID, fmt.Errorf("delete(%q, %q) failed; status %v: object %v still listed after delete", bucket, filename, status, obj.Name)
+		}
+		if err != nil {
+			status := m.BucketMissing
+			return fileID, fmt.Errorf("delete(%q, %q) failed; status %v: %w", bucket, filename, status, err)
+		}
+	}
+
+	// Update in-memory NIL file after delete operation.
+	delete(target.Journal.Filenames, int32(fileID))
+
+	logger.Infof("Object %v deleted in bucket %s.", file, bucket)
+	return fileID, nil
+}
+
+// pickFileToDelete picks which file to delete and returns the integer ID of this file.
+// Returns:
+//	- ID: returns the ID of the file to be deleted.
+func pickFileToDelete() int32 {
+	const (
+		begin                  = 11 // we can delete files staring from the file Hermes_11
+		numberOfDeletableFiles = 40 // there are 40 files to delete from [Hermes_11,Hermes_50]
+	)
+	rand.Seed(time.Now().UnixNano())
+	return int32(rand.Intn(numberOfDeletableFiles) + being)
```

Comment:
Fix typo.
Also, make sure code compiles before sending it for review.

Reply:
Typo fixed, apologies.